File: app/src/main/java/org/metabrainz/mobile/data/sources/api/entities/mbentity/Area.java
package org.metabrainz.mobile.data.sources.api.entities.mbentity;

import androidx.annotation.NonNull;

import com.google.gson.annotations.SerializedName;

import org.metabrainz.mobile.data.sources.api.entities.Alias;
import org.metabrainz.mobile.data.sources.api.entities.LifeSpan;

import java.util.ArrayList;
import java.util.List;

public class Area extends MBEntity {

    //TODO: ISO codes field to be added

    private String type;
    private String name;
    @SerializedName("sort-name")
    private String sortName;
    private final List<Alias> aliases = new ArrayList<>();
    @SerializedName("life-span")
    private LifeSpan lifeSpan;

    @NonNull
    @Override
    public String toString() {
        return "Area{" +
                "id='" + mbid + '\'' +
                ", type='" + type + '\'' +
                ", name='" + name + '\'' +
                ", sortName='" + sortName + '\'' +
                ", aliases=" + aliases +
                ", disambiguation='" + disambiguation + '\'' +
                ", lifeSpan=" + lifeSpan +
                '}';
    }

    public LifeSpan getLifeSpan() {
        return lifeSpan;
    }

    public void setLifeSpan(LifeSpan lifeSpan) {
        this.lifeSpan = lifeSpan;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getSortName() {
        return sortName;
    }

    public void setSortName(String sortName) {
        this.sortName = sortName;
    }

    public List<Alias> getAliases() {
        return aliases;
    }

    public void setAliases(List<Alias> aliases) {
        this.aliases.addAll(aliases);
    }
}

File: app/src/main/java/org/metabrainz/android/model/mbentity/Area.kt
package org.metabrainz.android.model.mbentity

import com.google.gson.annotations.SerializedName
import org.metabrainz.android.model.entities.LifeSpan
import java.util.*

class Area : org.metabrainz.android.model.mbentity.MBEntity() {
    //TODO: ISO codes field to be added
    var type: String? = null
    var name: String? = null

    @SerializedName("sort-name")
    var sortName: String? = null
    private val aliases: MutableList<org.metabrainz.android.model.entities.Alias> = ArrayList()

    @SerializedName("life-span")
    var lifeSpan: LifeSpan? = null
    override fun toString(): String {
        return "Area{" +
                "id='" + mbid + '\'' +
                ", type='" + type + '\'' +
                ", name='" + name + '\'' +
                ", sortName='" + sortName + '\'' +
                ", aliases=" + aliases +
                ", disambiguation='" + disambiguation + '\'' +
                ", lifeSpan=" + lifeSpan +
                '}'
    }

    fun getAliases(): List<org.metabrainz.android.model.entities.Alias> {
        return aliases
    }

    fun setAliases(aliases: List<org.metabrainz.android.model.entities.Alias>?) {
        this.aliases.addAll(aliases!!)
    }
}